src/docfx/lib/git/GitUtility.cs

```diff
@@ -118,9 +119,12 @@ public static bool IsRepo(string path)
         public static Task Clone(string cwd, string remote, string path, string branch = null, bool bare = false)
         {
             Directory.CreateDirectory(cwd);
+
+            // clone with configuration core.longpaths turned-on
+            // https://stackoverflow.com/questions/22575662/filename-too-long-in-git-for-windows#answer-40909460
             var cmd = string.IsNullOrEmpty(branch)
-                ? $"clone {remote} \"{path.Replace("\\", "/")}\""
-                : $"clone -b {branch} --single-branch {remote} \"{path.Replace("\\", "/")}\"";
+                ? $"clone -c core.longpaths=true {remote} \"{path.Replace("\\", "/")}\""
```

Comment:
can we add a spec test for that?

Reply:
+1

---
In reply to: [212568157](https://github.com/dotnet/docfx/pull/3282#discussion_r212568157) [](ancestors = 212568157)